### ISSUE TITLE
Support for default exports

### ISSUE THREAD

**Issue by coballast** [coballast/threejs-browserify-conversion-utility#16]
For modules that only export one thing, we want default exports and requires. See @kumavis's comments in #9 

I actually had this working at an earlier point in time in the source tree. I removed it because I felt like it vastly complicated the software, and would make diagnosing issues more difficult. Re-adding it should be a trivial exercise. See 22e2cbabfcc61b1f434f03a6834022155d21cd03

Many things have changed since then, but I think all we'd have to do is undo that change.

Whitewash my fence. It will be fun.

**Comment by kumavis** [comment]
Example:

```js
var Vector3 = require('./vector3.js')
```
not 
```js
var Vector3 = require('./vector3.js').Vector3
```

**Comment by kumavis** [comment]
This change needs to happen on the import AND export side

**Comment by HMUDesign** [comment]
For modules which export more than one thing, they should be split into more modules. There are some exceptions, such as util type libraries. I would recommend some sort of standard, such as every module exports a single default constructor, except files named *Util.js, which export an object containing the utility functions.

The last exception being the Three.js file used for producing the distributable build, but this should not be used except to build.

**Comment by coballast** [comment]
I agree that exporting only one thing is the ideal case. I am very pro Unix Philosophy (which THREE.js does not reflect). However, my main objective is to get a functional build working while changing the layout/structure as little as physically possible. The intention is (hopefully) to get it merged back with the mainline. After that's done, we can attack these structure issues separately.

Alternatively, if a pr never gets accepted, I'll just maintain a seperate fork. People will quickly realize that its better/less insane.

**Comment by coballast** [comment]
Default exports and imports is now working again. The code is not pretty, but it was the only way I got it working earlier. I tried to implement a cleaner and more concise approach using the acorn parser to generate the ast node structures for me, but I ended up having compatability issues with escodegen. I tried to kill two birds with one stone by using recast to both parse code, generate ast nodes, and generate output code. Had trouble there too.

I will elaborate my problems in more detail in #23 , but closing this for now. Feel free to open it back up if you have problems.
